kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -254,4 +270,25 @@ private void validate(String value) {
     private static void addConfig(HashMap<String, TableConfig<?>> configs, TableConfig<?> config) {
         configs.put(config.getKey().toLowerCase(Locale.ROOT), config);
     }
+
+    private static Map<String, String> parseJSONKeyValueMap(Engine engine, String jsonString) {
+        if (jsonString == null) {
+            return Collections.emptyMap();
+        }
+        StructType schema = new StructType()
+                .add("config", new MapType(StringType.STRING, StringType.STRING, true));
+        CloseableIterator<Row> batchRows = engine.getJsonHandler().parseJson(
+                singletonStringColumnVector(String.format("{\"config\": %s}", jsonString)),
```

Comment:
Please add a comment explaining how this hack enables reuse of the existing parser.